ipnsfs/file.go

```diff
@@ -0,0 +1,124 @@
+package ipnsfs
+
+import (
+	"sync"
+
+	chunk "github.com/jbenet/go-ipfs/importer/chunk"
+	dag "github.com/jbenet/go-ipfs/merkledag"
+	mod "github.com/jbenet/go-ipfs/unixfs/mod"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+)
+
+type File struct {
+	parent childCloser
+	fs     *Filesystem
+
+	name       string
+	hasChanges bool
+
+	mod  *mod.DagModifier
+	lock sync.Mutex
+}
+
+func NewFile(name string, node *dag.Node, parent childCloser, fs *Filesystem) (*File, error) {
+	dmod, err := mod.NewDagModifier(context.Background(), node, fs.dserv, fs.pins.GetManual(), chunk.DefaultSplitter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &File{
+		fs:     fs,
+		parent: parent,
+		name:   name,
+		mod:    dmod,
+	}, nil
+}
+
+func (fi *File) Write(b []byte) (int, error) {
+	fi.hasChanges = true
+	return fi.mod.Write(b)
+}
+
+func (fi *File) Read(b []byte) (int, error) {
+	fi.lock.Lock()
+	defer fi.lock.Unlock()
+	return fi.mod.Read(b)
+}
+
+func (fi *File) Close() error {
+	fi.lock.Lock()
+	defer fi.lock.Unlock()
+	if fi.hasChanges {
```

Comment:
this is a minor nit, but in general prefer a pattern like:

```go
if !fi.hasChanges {
  return nil
}

... the rest
```
than
```go

if fi.hasChanges {
  ... the rest
}
return nil
```

in the latter, there's more going on + more surface area for something to go wrong